invenio_github/tasks.py

```diff
@@ -144,13 +144,15 @@ def process_release(release_id, verify_sender=False):
     except Exception as ex:
         error_handlers = current_github.release_error_handlers
         release.model.status = ReleaseStatus.FAILED
-
         for error_cls, handler in error_handlers + DEFAULT_ERROR_HANDLERS:
             if isinstance(ex, error_cls):
                 handler(release, ex)
                 current_app.logger.exception(
                     u'Error while processing GitHub release')
                 break
+        if error_cls is Exception:
+            db.session.commit()
```

Comment:
The `finally` block does a `db.session.commit()` anyways, so this might not be necessary here... I see that there is a "timing" concern though, in case the retry task starts before the `db.session.commit()`.

Reply:
After reading the docs it's mentioned that the retry function is raising an Exception to stop the execution of the code. So it looks cleaner from that perspective than adding complex logic in the ```finally``` block.